libsolidity/ast/AST.cpp

```diff
@@ -407,19 +408,30 @@ FunctionDefinition const& FunctionDefinition::resolveVirtual(
 
 	FunctionType const* functionType = TypeProvider::function(*this)->asExternallyCallableFunction(false);
 
+	bool foundSearchStart = (_searchStart == nullptr);
 	for (ContractDefinition const* c: _mostDerivedContract.annotation().linearizedBaseContracts)
 	{
-		if (_searchStart != nullptr && c != _searchStart)
+		if (!foundSearchStart && c != _searchStart)
 			continue;
-		_searchStart = nullptr;
+		else
+			foundSearchStart = true;
+
 		for (FunctionDefinition const* function: c->definedFunctions())
 			if (
 				function->name() == name() &&
 				!function->isConstructor() &&
+				// Lookup via super should skip an unimplemented function because there must be
+				// another, implemented one behind it (otherwise we would fail at analysis stage).
+				// For ordinary virtual lookup, on the other hand, the only case in which we might
```

Comment:
So if this cannot happen for virtual functions, why not also skip unimplemented functions in that case?

Reply:
It can happen - in the one situation the comment mentions. I.e. when the call is in a function that will never run because it has been overridden. See for example: [call_unimplemented_base.sol](https://github.com/ethereum/solidity/blob/develop/test/libsolidity/semanticTests/functionCall/inheritance/call_unimplemented_base.sol).

My first implementation was actually what you suggested but it felt a bit like a hack. To cover this single case I had to remember the last seen unimplemented function and go back to it if it turned out there were no implemented ones at all. Doing it only for `super` makes it simpler.

Reply:
> super essentially searches upwards while virtual searches downwards

I do not understand  what you mean by this part. They both walk the linearized hierarchy in the same direction. Can you elaborate a bit more on how you see this situation?

Reply:
Comment changed.

Was:
```c++
// Lookup via super should skip an unimplemented function because there must be
// another, implemented one behind it (otherwise we would fail at analysis stage).
// For ordinary virtual lookup, on the other hand, the only case in which we might
// encounter an unimplemented function is when there is no implementation at all
// and that only happens if the code passed analysis due to the call never actually
// happening (i.e. it's inside a function that gets overridden).
```

is:
```c++
// With super lookup analysis guarantees that there is an implemented function in the chain.
// With virtual lookup there are valid cases where returning an unimplemented one is fine.
```